services/core-api/app/api/utils/models_mixins.py

```diff
@@ -198,32 +206,45 @@ def _deep_update_from_dict(self, data_dict, depth=0, _edit_key=None):
 
             if k in [c.name for c in editable_columns]:
                 col = next(col for col in editable_columns if col.name == k)
+
                 #get column definition for
                 current_app.logger.debug(depth * ' ' + f'updating {self}.{k}={v}')
+
                 if (type(col.type) == UUID):
                     #UUID does not implement python_type, manual check
                     assert isinstance(v, (UUID, str))
                 else:
                     py_type = col.type.python_type
+
                     if py_type == bool and not isinstance(v, bool) and v is not None:
                         raise DictLoadingError(
                             f"cannot assign '{k}':{v}{type(v)} to column of type {py_type}")
 
                     if py_type == datetime or py_type == date:
+                        #values that are considered null
+                        if v in [None, '']:
+                            setattr(self, k, None)
                         #json value is string, if expecting datetime in that column, convert here
-                        if v is not None:
+                        else:
                             setattr(self, k, parser.parse(v))
                         continue
+
                     if py_type == decimal.Decimal:
-                        #if Decimal column, cast whatever you get to Decimal
-                        dec = decimal.Decimal(v)
-                        #don't care about anything more precise, procection if incoming data is float
-                        setattr(self, k, dec.quantize(decimal.Decimal('.0000001')))
+                        #values that are considered null
+                        if v in [None, '']:
+                            setattr(self, k, None)
+                        else:
+                            #if Decimal column, cast whatever you get to Decimal
```

Comment:
Looks good, thanks for reworking this as discussed.

Reply:
Of course!

Reply:
Can you add to the deep_update testing suite this exact behaviour. (one happy like you have, and one nonsense like we talked about `description=[]` or something like that).

Reply:
Sure, I'll take a look.